test/peer-store/peer-store.spec.js

```diff
@@ -158,4 +158,57 @@ describe('peer-store', () => {
       expect(peerListenint4[1].id.toB58String()).to.eql(peerIds[3].toB58String())
     })
   })
+
+  describe('peerStore.peers', () => {
+    let peerStore
+
+    beforeEach(() => {
+      peerStore = new PeerStore()
+    })
+
+    it('returns peers if only have addresses', () => {
+      peerStore.addressBook.set(peerIds[0], [addr1])
+
+      const peers = peerStore.peers
+      expect(peers.size).to.equal(1)
+
+      const peerData = peers.get(peerIds[0].toB58String())
+      expect(peerData).to.exist()
+      expect(peerData.id).to.exist()
+      expect(peerData.addresses).to.have.lengthOf(1)
+      expect(peerData.protocols).to.have.lengthOf(0)
+      expect(peerData.metadata).to.not.exist()
+    })
+
+    it('returns peers if only have protocols', () => {
```

Comment:
This test is an interesting case to consider for the future. If we don't have addresses for a peer we may want to do some datastore cleanup, however there is a case for multiaddrs expiring and then we search peer routing to find the latest record for the peer because it previously supported a certain protocol. We should create an issue to address datastore cleanup strategies for future iterations.

```suggestion
    it('returns peers if only protocols are known', () => {
```

Reply:
I agree and I will create that issue